fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/mixin/dimension/LevelStorageMixin.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * Testing:
+ * A good way to test this is to create a world with the dimension test mod enabled and then without to simulate a dimension removal.
+ *
+ * Tracking issue: https://bugs.mojang.com/browse/MC-197860
+ */
+@Mixin(LevelStorage.class)
+abstract class LevelStorageMixin {
+	/*
+	 * When the custom dimension is removed, DimensionGeneratorSettings.CODEC fails to fully parse the Dynamic<T>,
+	 * so the DataResult<DimensionGeneratorSettings> only return the Either#Right part of the result,
+	 * which only includes the overworld.
+	 *
+	 * This causes all dimensions to cease to exist.
+	 * To fix this issue, we will intercept the dynamic being parsed, and remove any dimension entries that are NOT in the registry.
+	 */
+
+	/**
+	 * @param dynamic the dynamic containing the properties
+	 * @param dataFixer the data fixer
+	 * @param version the data version of the properties being read
+	 * @param info the callback info
+	 * @param propertiesDynamic the dynamic containing the world gen settings
+	 * @param <T> the type in the dynamic
+	 */
+	@Inject(method = "readGeneratorProperties", at = @At(value = "INVOKE", target = "Lcom/mojang/datafixers/DataFixer;update(Lcom/mojang/datafixers/DSL$TypeReference;Lcom/mojang/serialization/Dynamic;II)Lcom/mojang/serialization/Dynamic;"), locals = LocalCapture.CAPTURE_FAILHARD)
+	private static <T> void fixDimensionsOnLoad(Dynamic<T> dynamic, DataFixer dataFixer, int version, CallbackInfoReturnable<Pair<GeneratorOptions, Lifecycle>> info, Dynamic<T> propertiesDynamic) {
+		CompoundTag properties;
+
+		// Fix the dynamic in place if the value is nbt
+		if (dynamic.getValue() instanceof CompoundTag) {
+			properties = (CompoundTag) dynamic.getValue();
+		} else {
+			// Not supported atm
+			throw new IllegalStateException("Dynamic is not backed by a Compound Tag! This is not supported at the moment.");
+		}
+
+		// Get the dimension registry
+		DataResult<Pair<Registry<DimensionType>, T>> result = RegistryLookupCodec.of(Registry.DIMENSION_TYPE_KEY).codec().decode(propertiesDynamic);
+		Registry<DimensionType> dimensionRegistry = result.result().orElseThrow(() -> {
+			return new RuntimeException("Failed to get the dimension registry");
+		}).getFirst();
+
+		CompoundTag worldGenSettingsTag = properties.getCompound("WorldGenSettings");
+		CompoundTag dimensions = worldGenSettingsTag.getCompound("dimensions");
+
+		for (String key : dimensions.getKeys()) {
+			Identifier identifier;
+
+			try {
+				identifier = new Identifier(key);
+			} catch (InvalidIdentifierException e) {
+				throw new RuntimeException("Failed to fix generator properties because of invalid dimension identifier", e);
+			}
+
+			// Lookup the dimension
+			@Nullable
+			DimensionType dimension = dimensionRegistry.getOrEmpty(identifier).orElse(null);
+
+			// Dimension is not in registry, remove it from the compound tag so DFU does not freak out
+			if (dimension == null) {
```

Comment:
Why not just
```suggestion
			// Dimension is not in registry, remove it from the compound tag so DFU does not freak out
			if (!dimensionRegistry.containsId(identifier)) {
```

Reply:
See https://maven.fabricmc.net/docs/yarn-21w03a+build.23/net/minecraft/util/registry/Registry.html#containsId(net.minecraft.util.Identifier)